tests/pytests/vecsim_utils.py

```diff
@@ -0,0 +1,113 @@
+                        alg=alg,
+                        additional_vec_params=additional_vec_params,
+                        additional_schema_args=additional_schema_args)
+
+    populate_with_vectors(env, num_docs, dim, datatype, field_name)
+
+def get_tiered_debug_info(env, index_name, field_name) -> dict:
+    return to_dict(env.execute_command(debug_cmd(), "VECSIM_INFO", index_name, field_name))
+
+def get_tiered_frontend_debug_info(env, index_name, field_name) -> dict:
+    tiered_index_info = get_tiered_debug_info(env, index_name, field_name)
+    return to_dict(tiered_index_info['FRONTEND_INDEX'])
+
+def get_tiered_backend_debug_info(env, index_name, field_name) -> dict:
+    tiered_index_info = get_tiered_debug_info(env, index_name, field_name)
+    return to_dict(tiered_index_info['BACKEND_INDEX'])
+
+def get_vecsim_memory(env, index_key, field_name):
+    return float(to_dict(env.cmd(debug_cmd(), "VECSIM_INFO", index_key, field_name))["MEMORY"])/0x100000
+
+def get_vecsim_index_size(env, index_key, field_name):
+    return int(to_dict(env.cmd(debug_cmd(), "VECSIM_INFO", index_key, field_name))["INDEX_SIZE"])
+
+def get_redisearch_vector_index_memory(env, index_key):
+    return float(index_info(env, index_key)["vector_index_sz_mb"])
+
+def wait_for_background_indexing(env, index_name, field_name, message=''):
+    index_sizes = [0] * env.shardsCount
+    flat_index_sizes = [0] * env.shardsCount
+    backend_index_sizes = [0] * env.shardsCount
+    iter = 0
+    is_trained = [False] * env.shardsCount
+    index_state = f"iter: {iter}, index_sizes: {index_sizes}, flat_index_sizes: {flat_index_sizes}, backend_index_sizes: {backend_index_sizes}, is_trained: {is_trained}"
+
+    try:
+        with TimeLimit(120):
+            while not all(is_trained):
+                # 'BACKGROUND_INDEXING' == 0 means training is done
+                for i, con in enumerate(env.getOSSMasterNodesConnectionList()):
+                    tiered_info = get_tiered_debug_info(con, index_name, field_name)
+                    is_trained[i] = tiered_info['BACKGROUND_INDEXING'] == 0
+                    index_sizes[i] = tiered_info['INDEX_SIZE']
+                    flat_index_sizes[i] = to_dict(tiered_info['FRONTEND_INDEX'])['INDEX_SIZE']
+                    backend_index_sizes[i] = to_dict(tiered_info['BACKEND_INDEX'])['INDEX_SIZE']
+
+                time.sleep(0.1)
+                iter += 1
+                index_state = f"iter: {iter}, index_sizes: {index_sizes}, flat_index_sizes: {flat_index_sizes}, backend_index_sizes: {backend_index_sizes}, is_trained: {is_trained}"
+        for id, con in enumerate(env.getOSSMasterNodesConnectionList()):
+            index_size = get_tiered_debug_info(con, index_name, field_name)['INDEX_SIZE']
+            env.assertGreater(get_tiered_backend_debug_info(con, index_name, field_name)['INDEX_SIZE'], 0, message=f"wait_for_background_indexing: shard: {id}, index size: {index_size}" + message)
```

Comment:
Shouldn't we test that the backend index size is greater than the initial batch size?

Reply:
The purpose of this helper function is to verify that the background indexing was done
Stricter checks are done in the tests themselves
for example here https://github.com/RediSearch/RediSearch/blob/8b2b57b601dd60d409ca6e2f8750343cba9a9b38/tests/pytests/test_vecsim_svs.py#L148